docs/docs/tutorials/rag/langgraph_agentic_rag.ipynb

```diff
@@ -126,315 +227,567 @@
     "retriever_tool = create_retriever_tool(\n",
     "    retriever,\n",
     "    \"retrieve_blog_posts\",\n",
-    "    \"Search and return information about Lilian Weng blog posts on LLM agents, prompt engineering, and adversarial attacks on LLMs.\",\n",
-    ")\n",
-    "\n",
-    "tools = [retriever_tool]"
+    "    \"Search and return information about Lilian Weng blog posts.\",\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9cfdc027-5ea8-42ba-b0f4-0590505d76ff",
+   "metadata": {},
+   "source": [
+    "3\\. Test the tool:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "id": "58c17a90-9a85-46c9-96a9-1e7b224330c4",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "'(Note: Some work defines reward tampering as a distinct category of misalignment behavior from reward hacking. But I consider reward hacking as a broader concept here.)\\nAt a high level, reward hacking can be categorized into two types: environment or goal misspecification, and reward tampering.\\n\\nWhy does Reward Hacking Exist?#\\n\\nPan et al. (2022) investigated reward hacking as a function of agent capabilities, including (1) model size, (2) action space resolution, (3) observation space noise, and (4) training time. They also proposed a taxonomy of three types of misspecified proxy rewards:\\n\\nLet’s Define Reward Hacking#\\nReward shaping in RL is challenging. Reward hacking occurs when an RL agent exploits flaws or ambiguities in the reward function to obtain high rewards without genuinely learning the intended behaviors or completing the task as designed. In recent years, several related concepts have been proposed, all referring to some form of reward hacking:'"
+      ]
+     },
+     "execution_count": 9,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "retriever_tool.invoke({\"query\": \"types of reward hacking\"})"
    ]
   },
   {
    "cell_type": "markdown",
    "id": "fe6e8f78-1ef7-42ad-b2bf-835ed5850553",
    "metadata": {},
    "source": [
-    "## Agent State\n",
+    "## 3. Build the agentic RAG\n",
```

Comment:
I think we should get rid of this H2 and promote the H3s to numbered H2s. Can you incorporate this section's intro ("Now we will start building...") in the the next section ("Generate query")?

Reply:
sounds good!